### PR TITLE
Feature list players through endpoint

### PR DESCRIPTION
**Pull request by samBAM089**

### CODE REVIEW

File: backend/src/test/java/de/sambam/gamenighttracker/service/PlayerServiceTest.java [hunk 1]
```diff
@@ -36,4 +36,22 @@ public void addPlayerTester() {
...
+    @Test
+    @DisplayName("listAllPlayers should return all players")
+    public void listAllPlayersTest() {
+        //GIVEN
+        PlayerDb playerDb = new PlayerDb();
+        playerDb.add(Player.builder().id("001").name("samBAM").build());
+        playerDb.add(Player.builder().id("002").name("Andrea").build());
+
+        //WHEN
+        List<Player> playerList = playerDb.getPlayers();
```
Comment (christophersiem) [inline backend/src/test/java/de/sambam/gamenighttracker/service/PlayerServiceTest.java:49]: you are only testing your PlayerDB, not your PlayerService!

File: backend/src/test/java/de/sambam/gamenighttracker/service/PlayerServiceTest.java [hunk 2]
```diff
@@ -36,4 +36,22 @@ public void addPlayerTester() {
         assertTrue(playerList.contains(Player.builder().id("003").name("Thomas").build()));
 
     }
+
+    @Test
+    @DisplayName("listAllPlayers should return all players")
```
Comment (christophersiem) [inline backend/src/test/java/de/sambam/gamenighttracker/service/PlayerServiceTest.java:41]: you say that you want to test whether the method returns something, but in your //THEN block you only check if something is in your database

**Review by christophersiem — CHANGES_REQUESTED** [review]
Yay, keep it up 👍🏽 Good to see that you've implemented some tests. I left a couple of suggestions :)

File: backend/src/test/java/de/sambam/gamenighttracker/service/PlayerServiceTest.java (hunk 1 above)
Comment (samBAM089) [reply backend/src/test/java/de/sambam/gamenighttracker/service/PlayerServiceTest.java]: changes added : )

File: backend/src/test/java/de/sambam/gamenighttracker/controller/PlayerControllerTest.java [hunk 3]
```diff
@@ -35,10 +37,30 @@ public void resetDatabase() {
...
+    @Test
+    @DisplayName("get endpoint should show all players in db")
+    public void getAllPlayers() {
+        //GIVEN
+        playerDb.add(Player.builder().id("001").name("Richard").build());
+        playerDb.add(Player.builder().id("002").name("Tom").build());
+
+        //WHEN
+        ResponseEntity<Player[]> response = restTemplate.getForEntity(
+                "http://localhost:" + port + "players", Player[].class);
```
[the remark this answers is not in the document]
Comment (samBAM089) [reply backend/src/test/java/de/sambam/gamenighttracker/controller/PlayerControllerTest.java]: i tried to extract, but somehow my test did not work anymore... left it like this for now. whenever you got 5 mins, maybe we can have a look together ^^

File: backend/src/test/java/de/sambam/gamenighttracker/service/PlayerServiceTest.java (hunk 2 above)
Comment (samBAM089) [reply backend/src/test/java/de/sambam/gamenighttracker/service/PlayerServiceTest.java:41]: thanks for the hint. just made the respective changes ^^

**Review by christophersiem — APPROVED** [review]
lgtm :)

File: backend/src/test/java/de/sambam/gamenighttracker/controller/PlayerControllerTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (christophersiem) [reply backend/src/test/java/de/sambam/gamenighttracker/controller/PlayerControllerTest.java]: could you share a code snippet where you tried to extract the url?

[the remark this answers is not in the document]
Comment (samBAM089) [reply backend/src/test/java/de/sambam/gamenighttracker/controller/PlayerControllerTest.java]: I can try later. when in debug mode the port is set to null. when I don't extract it, the random port works fine...

**Pull request merged**